File: dashboard/src/utils/roles.js
// Single source of truth for role meaning + display in the dashboard.
//
// Mirrors normalizeRole() in the backend (src/employees/employees.service.ts):
// the dashboard reads Firestore directly and may see un-migrated legacy role
// values, so every screen classifies/labels roles through here rather than
// comparing raw strings.
//
// Canonical roles and their user-facing names:
//   office_employee  -> "Office employee"  (formerly Onsite Employee)
//   site_employee    -> "Site employee"    (formerly Offsite employee)
//   site_supervisor  -> "Site Supervisor"

export function normalizeRole(role) {
  if (role === 'siteAdmin' || role === 'site_supervisor') return 'site_supervisor'
  if (role === 'offsite_employee' || role === 'site_employee') return 'site_employee'
  return 'office_employee'
}

export const ROLE_LABELS = {
  office_employee: 'Office employee',
  site_employee: 'Site employee',
  site_supervisor: 'Site Supervisor',
}

export function roleLabel(role) {
  return ROLE_LABELS[normalizeRole(role)]
}

export function isSupervisorRole(role) {
  return normalizeRole(role) === 'site_supervisor'
}

export function isSiteEmployeeRole(role) {
  return normalizeRole(role) === 'site_employee'
}

File: mobile/lib/core/roles.dart
/// Canonical employee roles + normalization for the mobile app.
///
/// Mirrors normalizeRole() in the backend
/// (backend/src/employees/employees.service.ts). Records may still carry LEGACY
/// role values until the backend migration runs, and the app must keep working
/// for both, so every role decision in the UI goes through normalizeRole()
/// rather than comparing raw strings. Without this, a migrated "site_employee"
/// would fall through to the office default and lose its site screens.
///
/// User-facing names:
///   office_employee -> "Office employee" (formerly "Onsite Employee")
///   site_employee   -> "Site employee"   (formerly "Offsite employee")
///   site_supervisor -> "Site Supervisor"
library;

const String roleOfficeEmployee = 'office_employee';
const String roleSiteEmployee = 'site_employee';
const String roleSiteSupervisor = 'site_supervisor';

String normalizeRole(String? role) {
  if (role == 'siteAdmin' || role == 'site_supervisor') {
    return roleSiteSupervisor;
  }
  if (role == 'offsite_employee' || role == 'site_employee') {
    return roleSiteEmployee;
  }
  // onsite_employee / employee / null / anything else.
  return roleOfficeEmployee;
}

bool isSupervisorRole(String? role) =>
    normalizeRole(role) == roleSiteSupervisor;

bool isSiteEmployeeRole(String? role) =>
    normalizeRole(role) == roleSiteEmployee;

bool isOfficeEmployeeRole(String? role) =>
    normalizeRole(role) == roleOfficeEmployee;
